modules/src/clients/ics07_tendermint/client_state.rs

```diff
@@ -135,6 +150,45 @@ impl ClientState {
         Some(2 * self.trusting_period / 3)
     }
 
+    pub fn chain_id(&self) -> ChainId {
+        self.chain_id.clone()
+    }
+
+    pub fn client_type(&self) -> ClientType {
+        ClientType::Tendermint
+    }
+
+    pub fn client_def(&self) -> TendermintClient<G>
+    where
+        G: GlobalDefs,
+    {
+        TendermintClient::default()
+    }
+
+    pub fn frozen_height(&self) -> Option<Height> {
+        self.frozen_height
+    }
+
+    pub fn upgrade(
+        mut self,
+        upgrade_height: Height,
+        upgrade_options: UpgradeOptions,
+        chain_id: ChainId,
+    ) -> Self {
+        // Reset custom fields to zero values
+        self.trusting_period = ZERO_DURATION;
+        self.trust_level = TrustThreshold::ZERO;
+        self.frozen_height = None;
+        self.max_clock_drift = ZERO_DURATION;
```

Comment:
would be nice to wrap this into a helper function. Feels nicer to call `.reset()` :P